### PR TITLE
Replace flycheck-perl6 with flycheck-raku

### LINKED ISSUE #14804: Perl6 layer uses an obsolete package
#### Description :octocat:
perl6 layer requires flycheck-perl6 package which no longer exists

#### Reproduction guide :beetle:
- Enable perl6 layer
- Reload

*Observed behaviour:* :eyes: :broken_heart:
- See error message

*Expected behaviour:* :heart: :smile:
- No error message, flycheck supported in Raku (new name for Perl 6)

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. d9132f8bc)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-enable-snippets-in-popup t auto-completion-enable-sort-by-usage t)
 bm dap dtrt-indent emoji git gtags helm helpful imenu-list lsp multiple-cursors
 (spell-checking :variables spell-checking-enable-auto-dictionary t spell-checking-enable-by-default nil)
 syntax-checking templates treemacs typography version-control unicode-fonts ibuffer colors command-log copy-as-format dash debug deft
 (org :variables org-enable-org-journal-support t org-journal-enable-agenda-integration t org-journal-dir "~/org/journal" org-enable-roam-support t org-projectile-file "TODO.org")
 outshine pandoc pdf ansible asm c-c++ csv docker emacs-lisp go graphql graphviz html javascript json markdown nginx perl5 perl6 plantuml python rust scheme shell-scripts terraform yaml kappa olivetti)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND DBUS GSETTINGS GLIB NOTIFY INOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS XWIDGETS LIBSYSTEMD JSON PDUMPER GMP

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```

### PR DESCRIPTION
**Pull request by kappa**
flycheck-perl6 is obsolete and does not load.

This fixes #14804.

### CODE REVIEW

**Review by lebensterben — CHANGES_REQUESTED** [review]
Use something like this instead:

https://github.com/syl20bnr/spacemacs/blob/42dbd1d3597fb11f697fc80902472cbac836fe02/layers/%2Blang/rust/packages.el#L85

**Comment by kappa** [comment]
@lebensterben This is the closest I could get because the package does not define any functions, so there's nothing to autoload.

File: layers/+lang/perl6/packages.el [hunk 1]
```diff
@@ -26,26 +26,24 @@
...
 
 (defun perl6/post-init-flycheck ()
-  (spacemacs/enable-flycheck 'perl6-mode))
+  (spacemacs/enable-flycheck 'raku-mode))
 
-(defun perl6/init-flycheck-perl6 ()
-  (with-eval-after-load 'flycheck
-    (require 'flycheck-perl6)))
+(defun perl6/init-flycheck-raku ()
+  (add-hook 'flycheck-mode-hook (lambda () (use-package flycheck-raku))))
```
Comment (lebensterben) [inline layers/+lang/perl6/packages.el:45]: Unnecessary to `add-hook`
Just write
```elisp
(use-package flycheck-raku
  :defer t)
```

**Review by lebensterben — CHANGES_REQUESTED** [review]

**Comment by lebensterben** [comment]
A few other things:
- Rename the layer to raku
- Update the README.org and CHANGELOG.develop

File: layers/+lang/perl6/packages.el (hunk 1 above)
Comment (kappa) [reply layers/+lang/perl6/packages.el:45]: Thank you for the suggestion! Can you help me understand how the package will be loaded in this case? "Defer" will prevent it from loading on start and there are no autoloads set up for it.

Comment (lebensterben) [reply layers/+lang/perl6/packages.el:45]: okay I see

File: CHANGELOG.develop [hunk 2]
```diff
@@ -408,8 +408,8 @@ In org-agenda-mode
 - mercury (thanks to Ludvig Böklin)
 - pact (thanks to Colin Woodbury)
 - perl5 (thanks to Troy Hinckley, Jinseop Kim and Michael Rohleder)
-- perl6 (thanks to Bahtiar Gadimov and yuhan0)
 - prolog (thanks to Newres Al Haider)
+- raku (thanks to Bahtiar Gadimov and yuhan0)
```
Comment (lebensterben) [inline CHANGELOG.develop:412]: You can add your name here

**Review by lebensterben — CHANGES_REQUESTED** [review]
`raku` logo is licensed under Artistic 2.0, https://raw.githubusercontent.com/Raku/mu/master/misc/camelia.txt

You need to include a copy of its license to use it.

**Review by lebensterben — APPROVED** [review]

**Comment by lebensterben** [comment]
Merged. Thanks for your contribution.

**Pull request merged**